core/src/main/java/hudson/Functions.java

```diff
@@ -1895,6 +1895,17 @@ public String getServerName() {
         return Stapler.getCurrentRequest().getServerName();
     }
 
+    /**
+     * Returns the URL of the Jenkins.
+     * This method is different from {@link RunUrl#getBaseUrl()}.
+     * This method return Jenkins' base URL instead of only host of URL.
+     * For example, in development mode, Jenkins' base URL is "http://localhost:8080/jenkins" instead of "http://localhost:8080",
+     * thus, this method will return "http://localhost:8080/jenkins"
+     */
+    public String getRootUrl() {
+        return Jenkins.get().getRootUrl();
+    }
+
```

Comment:
Almost all Jelly/Groovy views have `rootURL` defined that should be (almost) the same value.

It's also not generally recommended to use this for the UI, as Jenkins may be accessed through different URLs (e.g. from different networks) and then the UI using this will break.

Reply:
I understand. But without this, the link will not work...